examples/typeclass-tests.dx

```diff
@@ -0,0 +1,10 @@
+interface TestInterface where
+  foo : Int
+
+instance TestInstance : TestInterface where
+  foo = 1
+> Parse error:5:10:
+>   |
+> 5 |   foo = 1
+>   |          ^
+> Interface constructor was not applied to any type and can not be instantiated
```

Comment:
Did we actually want to keep these instantiations though? Not that they're particularly useful, but I guess if we're going to forbid such an instantiation, it should also make sense to throw an error when declaring an interface without any parameter type. WDYT?

Reply:
Hmmm why would we disallow that? It looks like you could simulate implicit (in the sense of default values) arguments this way, so why not?

Reply:
I agree that there's no particular reason to forbid this, although I'm not sure how useful it would be.

Reply:
I misunderstood, I thought this was what you meant with your comment on #249. I also don't really think there are reasons for forbidding it. We all agree on that, then :)